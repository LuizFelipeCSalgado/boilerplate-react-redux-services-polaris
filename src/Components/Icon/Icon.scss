@import '../../../common/assets/variables.scss';

svg {
  vertical-align: middle;
}