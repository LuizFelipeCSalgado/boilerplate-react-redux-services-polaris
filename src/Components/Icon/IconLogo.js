import React from 'react';
import PropTypes from 'prop-types';

function IconLogo({ color }) {
  return (
    <div>
      {color === 'mono' ? (
        <svg width="47px" height="24px" viewBox="0 0 47 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink">
          <g id="Sprint-1" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
            <g id="02.-Solicitar-acesso---768-PX" transform="translate(-361.000000, -56.000000)" fill="#C4CDD5">
              <g id="Group" transform="translate(361.000000, 56.000000)">
                <path d="M39.9842299,20.6112676 C39.9842299,18.9106479 41.3683686,17.532507 43.075292,17.532507 C44.7825547,17.532507 46.1666934,18.9106479 46.1666934,20.6112676 C46.1666934,22.3112113 44.7825547,23.6893521 43.075292,23.6893521 C41.3683686,23.6893521 39.9842299,22.3112113 39.9842299,20.6112676 Z" id="Fill-3" />
                <polygon id="Fill-4" points="40.3090511 0.148394366 45.8425511 0.148394366 45.8425511 15.683493 40.3090511 15.683493" />
                <path d="M14.439438,20.723493 C13.8827956,22.2064225 12.4518175,23.2705352 10.775781,23.2705352 C9.04340146,23.2705352 7.55913504,22.1212394 7.09074088,20.5663099 L0.069580292,0.148394366 L5.92484672,0.148394366 C6.80834672,2.99053521 10.4577482,14.7545915 10.6675073,15.8717746 L10.9064562,15.8717746 C11.0965292,14.7299155 14.7598467,2.98749296 15.6480985,0.148394366 L21.5054015,0.148394366 L14.439438,20.723493 Z" id="Fill-5" />
                <path d="M30.2823613,0.148394366 L30.2823613,17.9861408 L37.9246533,17.9861408 L37.9246533,23.2705352 L27.000208,23.2705352 C25.524427,23.2705352 24.3252701,22.0786479 24.3252701,20.6112676 L24.3252701,0.148394366 L30.2823613,0.148394366 Z" id="Fill-6" />
              </g>
            </g>
          </g>
        </svg>
      )
        : (
          <svg width="47px" height="24px" viewBox="0 0 47 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink">
            <g id="Symbols" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
              <g id="05-Sidenav-/-_Resources-/-Logo" transform="translate(0.000000, -3.000000)">
                <g id="Group" transform="translate(0.000000, 3.000000)">
                  <path d="M39.9842299,20.6112676 C39.9842299,18.9106479 41.3683686,17.532507 43.075292,17.532507 C44.7825547,17.532507 46.1666934,18.9106479 46.1666934,20.6112676 C46.1666934,22.3112113 44.7825547,23.6893521 43.075292,23.6893521 C41.3683686,23.6893521 39.9842299,22.3112113 39.9842299,20.6112676 Z" id="Fill-3" fill="#DE6C27" />
                  <polygon id="Fill-4" fill="#FFFFFF" points="40.3090511 0.148394366 45.8425511 0.148394366 45.8425511 15.683493 40.3090511 15.683493" />
                  <path d="M14.439438,20.723493 C13.8827956,22.2064225 12.4518175,23.2705352 10.775781,23.2705352 C9.04340146,23.2705352 7.55913504,22.1212394 7.09074088,20.5663099 L0.069580292,0.148394366 L5.92484672,0.148394366 C6.80834672,2.99053521 10.4577482,14.7545915 10.6675073,15.8717746 L10.9064562,15.8717746 C11.0965292,14.7299155 14.7598467,2.98749296 15.6480985,0.148394366 L21.5054015,0.148394366 L14.439438,20.723493 Z" id="Fill-5" fill="#FFFFFF" />
                  <path d="M30.2823613,0.148394366 L30.2823613,17.9861408 L37.9246533,17.9861408 L37.9246533,23.2705352 L27.000208,23.2705352 C25.524427,23.2705352 24.3252701,22.0786479 24.3252701,20.6112676 L24.3252701,0.148394366 L30.2823613,0.148394366 Z" id="Fill-6" fill="#FFFFFF" />
                </g>
              </g>
            </g>
          </svg>
        )
      }
    </div>
  );
}
IconLogo.propTypes = {
  color: PropTypes.string,
};
export default IconLogo;
