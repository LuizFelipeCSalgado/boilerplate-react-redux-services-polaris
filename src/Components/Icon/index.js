import React from 'react';
import PropTypes from 'prop-types';
import { icons } from 'Common/Helpers/IconHelper';
import { colors } from 'Config/Palette';

import './Icon.scss';

function Icon({
  path, size, color, transform,
}) {
  const cor = colors[color] === undefined ? color : colors[color];
  const sizeDefault = size === undefined ? '21px' : size;
  const pointsPolygon = [
    ' 40.3090511 0.148394366 45.8425511 0.148394366 ',
    ' 45.8425511 15.683493 40.3090511 15.683493 ',
  ];
  return (
    <svg
      fill={cor}
      width={sizeDefault}
      height={sizeDefault}
      viewBox="0 0 24 24"
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
      style={{
        alignSelf: 'center',
        transform: transform || null,
      }}
    >
      <g id={path} stroke="none" strokeWidth="1" fillRule="evenodd">
        <path d={icons[path]} id="Icon" />
        {path === 'logo1' ? (
          <polygon id="Fill-4" points={`${pointsPolygon[0]} ${pointsPolygon[1]}`} />
        ) : (
          false
        )}
      </g>
    </svg>
  );
}

Icon.propTypes = {
  path: PropTypes.string,
  size: PropTypes.string,
  color: PropTypes.string,
  transform: PropTypes.string,
};

Icon.defaultProps = {
  size: '24px',
  color: 'red',
  transform: '',
  path: '',
};

export default Icon;
