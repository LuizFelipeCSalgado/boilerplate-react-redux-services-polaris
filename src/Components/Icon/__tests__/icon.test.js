import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Icon from '..';

describe('Icon Test', () => {
  const props = {
    path: 'home',
  };
  const wrapper = shallow(<Icon {...props} />);

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with svg element', () => {
    expect(wrapper.find('svg')).toHaveLength(1);
  });

  it('should mount svg element fill, width, and height default attributes', () => {
    expect(wrapper.find('svg[fill="red"]')).toHaveLength(1);
    expect(wrapper.find('svg[width="24px"]')).toHaveLength(1);
    expect(wrapper.find('svg[height="24px"]')).toHaveLength(1);
  });

  it('should mount svg element fill, width, and height passed attributes', () => {
    const newWrapper = shallow(<Icon {...props} size="40px" color="darkGray" />);
    expect(newWrapper.find('svg[fill="#212B36"]')).toHaveLength(1);
    expect(newWrapper.find('svg[width="40px"]')).toHaveLength(1);
    expect(newWrapper.find('svg[height="40px"]')).toHaveLength(1);
  });
});
