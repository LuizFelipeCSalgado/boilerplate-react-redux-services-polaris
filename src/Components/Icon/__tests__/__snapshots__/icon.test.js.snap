// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Test renders correctly 1`] = `
<svg
  fill="red"
  height="24px"
  style={
    Object {
      "alignSelf": "center",
      "transform": null,
    }
  }
  version="1.1"
  viewBox="0 0 24 24"
  width="24px"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <g
    fillRule="evenodd"
    id="home"
    stroke="none"
    strokeWidth="1"
  >
    <path
      d="M21.6637292,10.2522 L12.6637292,2.25219995 C12.2847292,1.91619995 11.7147292,1.91619995 11.3357292,2.25219995 L9.99972917,3.44019996 L9.99972917,3.00019995 C9.99972917,2.44819995 9.55272917,2.00019995 8.99972916,2.00019995 L4.99972914,2.00019995 C4.44672914,2.00019995 3.99972914,2.44819995 3.99972914,3.00019995 L3.99972914,8.77319998 L2.33572913,10.2522 C1.92272913,10.6202 1.88572913,11.2512 2.25272913,11.6652 C2.61872913,12.0772 3.25072914,12.1122 3.66372914,11.7482 L3.99972914,11.4492 L3.99972914,21.0002 C3.99972914,21.5522 4.44672914,22.0002001 4.99972914,22.0002001 L18.9997292,22.0002001 C19.5527292,22.0002001 19.9997292,21.5522 19.9997292,21.0002 L19.9997292,11.4492 L20.3357292,11.7482 C20.5257292,11.9172 20.7637292,12.0002 20.9997292,12.0002 C21.2747292,12.0002 21.5497292,11.8872 21.7467292,11.6652 C22.1137293,11.2512 22.0767293,10.6202 21.6637292,10.2522 Z M17.9997292,20.0002 L15.9997292,20.0002 L15.9997292,15.0002 C15.9997292,14.4482 15.5527292,14.0002 14.9997292,14.0002 L8.99972916,14.0002 C8.44672916,14.0002 7.99972916,14.4482 7.99972916,15.0002 L7.99972916,20.0002 L5.99972915,20.0002 L5.99972915,9.67119999 L11.9997292,4.33819996 L17.9997292,9.67119999 L17.9997292,20.0002 Z M9.99972917,20.0002 L13.9997292,20.0002 L13.9997292,16.0002 L9.99972917,16.0002 L9.99972917,20.0002 Z M5.99972915,4.00019996 L7.99972916,4.00019996 L7.99972916,5.21819997 L5.99972915,6.99619997 L5.99972915,4.00019996 Z"
      id="Icon"
    />
  </g>
</svg>
`;
